kernel/runtime/cxx/runtime/futex.cc

```diff
@@ -167,6 +169,85 @@ static int futex_wake(
+    MLOG_DETAIL(mlog::app, "Futex_requeue", DVARhex(uaddr), DVARhex(uaddr2), DVAR(val), DVAR(val2));
+    uint32_t hashFrom = hash_futex(uaddr);
+
+    // wakes up a maximum of  val  waiters that  are  waiting  on  the  futex at uaddr
+    mythos::Mutex::Lock guard0(requeueLock);
+    mythos::Mutex::Lock guard1(queue[hashFrom].readLock);
+    auto curr = &(queue[hashFrom].queueHead);
+    for (int i = 0; i < val; i++) {
+        while (curr->load(std::memory_order_relaxed) && curr->load(std::memory_order_relaxed)->uaddr != uaddr) {
+            MLOG_DETAIL(mlog::app, "Skip entry", DVARhex(curr->load(std::memory_order_relaxed)->uaddr),
+                        DVAR(curr->load(std::memory_order_relaxed)->ec), DVAR(curr->load(std::memory_order_relaxed)->queueNext), DVARhex(uaddr) );
+            curr = &curr->load(std::memory_order_relaxed)->queueNext;
+        }
+
+        if (curr->load(std::memory_order_relaxed) == nullptr){
+          MLOG_DETAIL(mlog::app, "requeue end (end of queue reached)" );
+          return 0;
+        }
+
+        if (queue[hashFrom].queueTail == &curr->load(std::memory_order_relaxed)->queueNext) {
+            // own element is the tail of list:
+            // move tail to the element that points to our element
+            MLOG_DETAIL(mlog::app, "last elem in queue");
+            queue[hashFrom].queueTail = curr;
+        }
+        MLOG_DETAIL(mlog::app, uaddr, "Found entry" );
+        auto entry = curr->load(std::memory_order_relaxed);
+        curr->store(entry->queueNext, std::memory_order_relaxed);
+        auto ec = entry->ec;
+        entry->queueNext.store(reinterpret_cast<FutexQueueElem*>(1ul)); // mark as removed
+
+        MLOG_DETAIL(mlog::app, "Wake EC", DVAR(ec), DVAR(uaddr) );
+        mythos::syscall_signal(ec);
+    }
+
+    // If there are more than val waiters, then the remaining
+    // waiters are removed from the wait queue of the source futex at uaddr and added to the wait queue  of
+    // the  target  futex  at  uaddr2.  The val2 argument specifies an upper limit on the number of waiters
+    // that are requeued to the futex at uaddr2.
+    {
+      uint32_t hashTo = hash_futex(uaddr2);
+      mythos::Mutex::Lock guard2(queue[hashTo].readLock);
+      MLOG_DETAIL(mlog::app, "requeue waiters", DVAR(hashTo) );
+      for (int i = 0; i < val2; i++) {
+          while (curr->load(std::memory_order_relaxed) && curr->load(std::memory_order_relaxed)->uaddr != uaddr) {
+              MLOG_DETAIL(mlog::app, "Skip entry", DVARhex(curr->load(std::memory_order_relaxed)->uaddr),
+                          DVAR(curr->load(std::memory_order_relaxed)->ec), DVAR(curr->load(std::memory_order_relaxed)->queueNext), DVARhex(uaddr) );
+              curr = &curr->load(std::memory_order_relaxed)->queueNext;
+          }
+
+          if (curr->load(std::memory_order_relaxed) == nullptr){
+            MLOG_DETAIL(mlog::app, "requeue end (end of queue reached)" );
+            return 0;
+          }
```

Comment:
Since this is the second time we use that loop, maybe just extract it into a named function 🤔

Reply:
added comment

Reply:
it is a loop of only two lines of code. the remaining line are for debug output.